lib/wallaroo/core/source/connector_source/connector_framed_source_notify.pony

```diff
@@ -16,24 +16,155 @@ Copyright 2017-2019 The Wallaroo Authors.
 
 */
 
+use "backpressure"
+use "buffered"
 use "collections"
-use "time"
-use "wallaroo_labs/time"
+use "promises"
 use "wallaroo/core/boundary"
 use "wallaroo/core/common"
-use "wallaroo/core/partitioning"
+use "wallaroo/core/checkpoint"
 use "wallaroo/core/data_receiver"
-use "wallaroo/core/recovery"
-use "wallaroo_labs/mort"
+use "wallaroo/core/invariant"
 use "wallaroo/core/messages"
 use "wallaroo/core/metrics"
+use "wallaroo/core/partitioning"
+use "wallaroo/core/recovery"
 use "wallaroo/core/routing"
 use "wallaroo/core/source"
 use "wallaroo/core/topology"
+use "wallaroo_labs/bytes"
+use cwm = "wallaroo_labs/connector_wire_messages"
+use "wallaroo_labs/mort"
+use "wallaroo_labs/time"
+
+type _ProtoFsmState is (_ProtoFsmConnected | _ProtoFsmHandshake |
+                        _ProtoFsmStreaming | _ProtoFsmError |
+                        _ProtoFsmDisconnected | _ProtoFsmShrinking)
+primitive _ProtoFsmConnected
+  fun apply(): U8 => 1
```

Comment:
Are these `apply()` methods used?

Reply:
Yes, they're used for `@printf` diagnostic messages immediately before calling `Fail()`.